be/src/exprs/agg/window_funnel.h

```diff
@@ -68,12 +73,15 @@ struct WindowFunnelState {
 
     // first args is timestamp, second is event position.
     using TimestampEvent = std::pair<TimestampType, uint8_t>;
-    mutable std::vector<TimestampEvent> events_list;
     using TimestampVector = std::vector<TimestampType>;
     int64_t window_size;
-    int32_t mode;
+    int32_t mode = 0;
     uint8_t events_size;
     bool sorted = true;
+    char buffer[64];
```

Comment:
why buffer size is 64?

Reply:
instead with  `char buffer[reserve_list_size * sizeof(TimestampEvent)];`